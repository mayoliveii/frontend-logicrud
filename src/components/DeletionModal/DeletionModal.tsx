import { Button, AlertDialog, AlertDialogOverlay, AlertDialogContent, AlertDialogHeader, AlertDialogCloseButton, AlertDialogBody, AlertDialogFooter } from "@chakra-ui/react";
import React, { useState } from "react";

interface DeletionModalProps {
  isOpen: boolean;
  onClose: () => void;
  onConfirm: () => void;
}

export const DeletionModal: React.FC<DeletionModalProps> = ({ isOpen, onClose, onConfirm }) => {

  const [isLoading, setIsLoading] = useState(false);
  const cancelRef = React.useRef<HTMLButtonElement>(null);

  const handleConfirm = async () => {
    setIsLoading(true);
    await onConfirm();
    setIsLoading(false);
  };

  return (
    <>
      <AlertDialog
        motionPreset='slideInBottom'
        leastDestructiveRef={cancelRef}
        onClose={onClose}
        isOpen={isOpen}
        isCentered
      >
        <AlertDialogOverlay />
        <AlertDialogContent>
          <AlertDialogHeader>Tem certeza que deseja deletar esta categoria?</AlertDialogHeader>
          <AlertDialogCloseButton />
          <AlertDialogBody>
            Essa ação é irreversível e não poderá ser desfeita. Deseja continuar?
          </AlertDialogBody>
          <AlertDialogFooter>
            <Button ref={cancelRef} onClick={onClose}>
              cancelar
            </Button>
            <Button
              isLoading={isLoading}
              bg={'#FF7A50'}
              _hover={{ bg: '#FF5823' }}
              color={'white'}
              ml={3}
              onClick={handleConfirm}
            >
              Sim
            </Button>
          </AlertDialogFooter>
        </AlertDialogContent>
      </AlertDialog>
    </>
  )
} 