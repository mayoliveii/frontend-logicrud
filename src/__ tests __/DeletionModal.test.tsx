import { render, fireEvent } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import '@testing-library/jest-dom';
import { DeletionModal } from '../components/DeletionModal/DeletionModal';

describe('DeletionModal component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <ChakraProvider>
        <DeletionModal isOpen={true} onClose={jest.fn()} onConfirm={jest.fn()} />
      </ChakraProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders when isOpen is true', () => {
    const { getByText } = render(
      <ChakraProvider>
        <DeletionModal isOpen={true} onClose={jest.fn()} onConfirm={jest.fn()} />
      </ChakraProvider>
    );
    expect(getByText('Tem certeza que deseja deletar esta categoria?')).toBeInTheDocument();
  });

  it('does not render when isOpen is false', () => {
    const { queryByText } = render(
      <ChakraProvider>
        <DeletionModal isOpen={false} onClose={jest.fn()} onConfirm={jest.fn()} />
      </ChakraProvider>
    );
    expect(queryByText('Tem certeza que deseja deletar esta categoria?')).not.toBeInTheDocument();
  });

  it('calls onClose when cancel button is clicked', () => {
    const onClose = jest.fn();
    const { getByText } = render(
      <ChakraProvider>
        <DeletionModal isOpen={true} onClose={onClose} onConfirm={jest.fn()} />
      </ChakraProvider>
    );
    fireEvent.click(getByText('cancelar'));
    expect(onClose).toHaveBeenCalled();
  });

  it('calls onConfirm when confirm button is clicked', () => {
    const onConfirm = jest.fn();
    const { getByText } = render(
      <ChakraProvider>
        <DeletionModal isOpen={true} onClose={jest.fn()} onConfirm={onConfirm} />
      </ChakraProvider>
    );
    fireEvent.click(getByText('Sim'));
    expect(onConfirm).toHaveBeenCalled();
  });
});