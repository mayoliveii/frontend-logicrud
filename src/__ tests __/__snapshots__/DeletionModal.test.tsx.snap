// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletionModal component matches snapshot 1`] = `
<DocumentFragment>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
